cypress/e2e/facility_spec/FacilityHomepage.cy.ts

```diff
@@ -181,6 +221,99 @@ describe("Facility Homepage Function", () => {
+    loginPage.ensureLoggedIn();
+  });
+
+  it("Verify the bed capacity badge reflection", () => {
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    manageUserPage.assertFacilityInCard(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("0", "0");
+
+    // link dummy doctor 12 to the facility
+    cy.awaitUrl("/users");
+    userPage.typeInSearchInput(usernameToLinkFacilityWithNoBeds);
+    userPage.checkUsernameText(usernameToLinkFacilityWithNoBeds);
+
+    cy.get("#home_facility").then(($homeFacility) => {
+      const homeFacilityText = $homeFacility.text().trim();
+      if (homeFacilityText.includes("No Home Facility")) {
+        // Link facility if no home facility exists
+        manageUserPage.clickFacilitiesTab();
+        manageUserPage.selectFacilityFromDropdown(facilityWithNoAvailableBeds);
+        manageUserPage.clickLinkFacility();
+        manageUserPage.clickHomeFacilityIcon();
+        manageUserPage.assertnotLinkedFacility(facilityWithNoAvailableBeds);
+        manageUserPage.assertHomeFacilitylink(facilityWithNoAvailableBeds);
+        manageUserPage.clickCloseSlideOver();
+      } else {
+        // Assert if facility is already linked
+        manageUserPage.assertHomeFacility(facilityWithNoAvailableBeds);
+      }
+    });
+
+    // create a new patient in the facility
+    cy.visit("/patients");
+    patientPage.createPatientWithData(newPatientData);
+    // navigate to facility page and verify the occupancy badge
+    cy.visit("/facility");
+    manageUserPage.typeFacilitySearch(facilityWithNoAvailableBeds);
+    facilityPage.verifyFacilityBadgeContent(facilityWithNoAvailableBeds);
+    facilityHome.verifyOccupancyBadgeVisibility();
+    manageUserPage.assertFacilityBadgeContent("1", "0");
+    manageUserPage.assertFacilityBadgeBackgroundColor("rgb(239, 68, 68)");
+    // create a new location and add a bed to the facility
+    facilityPage.visitAlreadyCreatedFacility();
+    cy.get("[id='manage-facility-dropdown']").scrollIntoView().click();
+    cy.get("[id=location-management]").click();
+    // create new location and add a bed to the facility
+    cy.get("body").then(($body) => {
+      if ($body.find("#manage-bed-button").length > 0) {
+        facilityLocation.clickManageBedButton();
+      } else {
+        facilityLocation.clickAddNewLocationButton();
+        facilityPage.fillFacilityName(locationName);
+        facilityLocation.selectLocationType(locationType);
+        assetPage.clickassetupdatebutton();
+        facilityLocation.clickNotification();
+        facilityLocation.clickManageBedButton();
+      }
+    });
+    facilityLocation.clickAddBedButton();
+    facilityLocation.addBed("Bed 1", "Test Description", "Regular", 2);
+    // navigate to patient page, and click create consultation
+    cy.visit("/facility");
```

Comment:
- Rather than redirecting to the facility page, typing and verifying the card again, it would be better to save the current page query parameter in a variable, when you are verifying the card capacity in the first time itself,  and use query-parameter-based redirection in this step.